ethereum/core/src/main/java/org/hyperledger/besu/ethereum/chain/DefaultBlockchain.java

```diff
@@ -70,12 +70,21 @@ private DefaultBlockchain(
       final BlockchainStorage blockchainStorage,
       final MetricsSystem metricsSystem,
       final long reorgLoggingThreshold) {
+    this(genesisBlock, blockchainStorage, metricsSystem, reorgLoggingThreshold, null);
+  }
+
+  private DefaultBlockchain(
+      final Optional<Block> genesisBlock,
+      final BlockchainStorage blockchainStorage,
+      final MetricsSystem metricsSystem,
+      final long reorgLoggingThreshold,
+      final String dataDirectory) {
     checkNotNull(genesisBlock);
     checkNotNull(blockchainStorage);
     checkNotNull(metricsSystem);
 
     this.blockchainStorage = blockchainStorage;
-    genesisBlock.ifPresent(this::setGenesis);
+    genesisBlock.ifPresent(block -> this.setGenesis(block, dataDirectory));
```

Comment:
it seems weird for the DefaultBlockchain to have the data directory?

Reply:
It doesn't, but since we handle the validation of the genesis block at this level we pass the string in as a param

Reply:
ok so it gets passed in and then discarded